src/ess/powder/conversion.py

```diff
@@ -211,11 +218,44 @@ def convert_to_dspacing(
         for key in ('wavelength', 'two_theta'):
             if key in out.coords.keys():
                 out.coords.set_aligned(key, False)
-    out.bins.coords.pop('tof', None)
+    # out.bins.coords.pop('tof', None)
     out.bins.coords.pop('wavelength', None)
     return DspacingData[RunType](out)
 
 
+def assemble_output_calibration(
+    data: DspacingData[SampleRun],
+) -> OutputCalibrationData:
+    # Use nanmean because pixels without events have position=NaN.
+    average_l = sc.nanmean(data.coords["Ltotal"])
+    average_two_theta = sc.nanmean(data.coords["two_theta"])
+    difc = sc.to_unit(
+        2
+        * sc.constants.m_n
+        / sc.constants.h
+        * average_l
+        * sc.sin(0.5 * average_two_theta),
+        unit='us / angstrom',
+    ).value
+    return OutputCalibrationData(
+        sc.DataArray(
+            sc.array(dims=['calibration'], values=[difc]),
+            coords={'power': sc.array(dims=['calibration'], values=[1])},
+        )
+    )
+
+
+def convert_to_calibrated_tof(
+    data: IofDspacing, calibration: OutputCalibrationData
+) -> IofTof:
+    difc = calibration['power', sc.scalar(1)].data
```

Comment:
Is the plan to have other powers as well? Is a `DataArray` the right data structure here? Maybe a `DataGroup` or simple `dataclass` would be more intuitive?

Reply:
> Is the plan to have other powers as well?

Ultimately, yes, as far as I know. @celinedurniak ?

See https://github.com/scipp/essdiffraction/pull/106#discussion_r1843367827 why it is a data array. A data group would not work because the keys should be powers (`int`) and not names (`str`). A dataclass would have a fixed set of fields. Plus it would also need names instead of powers.

Reply:
@jl-wynen your assumption is correct. There could be, for example, these terms to consider
difc * dspacing + difa * dispacing **2 + difb/dspacing +Zero

Reply:
Ok. Do we need to support them now or can this wait until we actually need them?